Can't find element in the rewritten iframe via Selector.withText

After switching to the iframe with dynamically created document, TestCafe can't find the existing element:
```js
AssertionError: expected false to be truthy

      + expected - actual

      -false
      +undefined


      Browser: Chrome 87.0.4280.141 / Windows 10 

          3 |fixture`Test`.page`http://localhost:3000/`;
          4 |
          5 |test('test', async t => {
          6 |    await t
          7 |        .switchToIframe('#frame')
       >  8 |        .expect(Selector('#test').withText('The Test Text').exists).ok();
          9 |});
         10 |
```
Although the element described by such selector present in the iframe. But if we replace "withText" method with "withExactText" then the test will pass successfully.

[testcafe-iframe-bug.zip](https://github.com/DevExpress/testcafe/files/5850395/testcafe-iframe-bug.zip)

--- a/src/client/driver/command-executors/client-functions/eval-function.js
+++ b/src/client/driver/command-executors/client-functions/eval-function.js
@@ -4,11 +4,17 @@ import { Promise, nativeMethods } from '../../deps/hammerhead';
 // NOTE: evalFunction is isolated into a separate module to
 // restrict access to TestCafe intrinsics for the evaluated code.
 // It also accepts `__dependencies$` argument which may be used by evaluated code.
-/* eslint-disable @typescript-eslint/no-unused-vars */
 export default function evalFunction (fnCode, __dependencies$) {
-    // NOTE: `eval` in strict mode will not override context variables
-    const evaluator = new nativeMethods.Function('fnCode', '__dependencies$', 'Promise', '"use strict"; return eval(fnCode)');
+    const evaluator = new nativeMethods.Function(
+        'fnCode',
+        '__dependencies$',
+        'Promise',
+        // NOTE: we should pass the original `RegExp`
+        // to make the `instanceof RegExp` check successful in different contexts
+        'RegExp',
+        // NOTE: `eval` in strict mode will not override context variables
+        '"use strict"; return eval(fnCode)'
+    );
 
-    return evaluator(fnCode, __dependencies$, Promise);
+    return evaluator(fnCode, __dependencies$, Promise, RegExp);
 }
-/* eslint-enable @typescript-eslint/no-unused-vars */

--- a/test/functional/fixtures/regression/gh-5886/pages/index.html
+++ b/test/functional/fixtures/regression/gh-5886/pages/index.html
@@ -1,0 +1,17 @@
+<!DOCTYPE html>
+<html>
+<head>
+    <meta charset="UTF-8">
+    <title>gh-5886</title>
+</head>
+<body>
+<iframe id="frame"></iframe>
+<br />
+<script>
+    var iframe = document.getElementById('frame');
+    iframe.contentDocument.open();
+    iframe.contentDocument.write('<html><body><div id="test">The Test Text</div></body></html>');
+    iframe.contentDocument.close();
+</script>
+</body>
+</html>

--- a/test/functional/fixtures/regression/gh-5886/test.js
+++ b/test/functional/fixtures/regression/gh-5886/test.js
@@ -1,0 +1,5 @@
+describe('[Regression](GH-5886) - Selector for element after switching to the rewritten iframe', function () {
+    it('Should exist', function () {
+        return runTests('testcafe-fixtures/index.js');
+    });
+});

--- a/test/functional/fixtures/regression/gh-5886/testcafe-fixtures/index.js
+++ b/test/functional/fixtures/regression/gh-5886/testcafe-fixtures/index.js
@@ -1,0 +1,10 @@
+import { Selector } from 'testcafe';
+
+fixture `Selector for element after switching to the rewritten iframe`
+    .page `http://localhost:3000/fixtures/regression/gh-5886/pages/index.html`;
+
+test('Should exist', async t => {
+    await t
+        .switchToIframe('#frame')
+        .expect(Selector('#test').withText('The Test Text').exists).ok();
+});

